### DISCUSSION TITLE
Trying to build SDR++ on Mac M1 for use with SDRPlay - can't install SDRPlay API

### DISCUSSION THREAD

**Discussion by vpnels** [AlexandreRouma/SDRPlusPlus#692] [Q&A]
I have searched the other topics, and I can't find an answer to this problem. I am trying to compile and build SDR++ on my MacBook Air M1 running Monterey 12.3. One of the prerequisites for the build is the SDRplay M1 API. I have downloaded the file SDRplay_RSP_API-MacOS-3.07.a4.run from the SDRplay.com website.

However, it will NOT install. The installer is not able to create the directory /usr/sdrplay and complete the installation. The error is "Operation not permitted". So, I went into the Mac's recovery mode and disabled the SIP (as per numerous postings on the web). After doing this, I still can't create the directory /usr/sdrplay because I receive the message "read-only file system". I am not able to get around this even directly with root privileges in the terminal.

Is there a way forward?

Pascal

**Comment by vpnels** [comment]
Progress.

Manually creating the directories

/usr/local/bin
/usr/local/include
/usr/local/lib

before running the API installer script allows it to install in the proper place.

**Comment by vpnels** [comment]
Manually adding the /usr/local libraries (prevous post) allowed me to build and run sdrpp on my MacBook Air M1. However, I do not have any audio sinks - so can't hear the audio. I used the same OPT_BUILD list (CMakelists.txt) as I used on my X64 iMac - which works. But still no audio sinks showing up. Just "none" and "network".

I can run the radio and see stations on the waterfall - just no audio.

**Reply by vpnels** [reply]
Got it. Didn't have an audio module configured. Now working on my MacBook Air M1.

**Reply by mrkva** [reply]
Hey, would you mind sharing the build? I am struggling with a very similar thing right now :) Thanks

**Reply by vpnels** [reply]
mrvka,

---
Build SDR++ for Mac M1 for SDRPlay source

See instructions at https://github.com/AlexandreRouma/SDRPlusPlus

Download and install SDRPlay API for M1. You must manually create the /usr/local/bin /usr/local/include /usr/local/lib directories before attempting to install the SDRplay API.

brew install cmake volk glew glfw airspy airspyhf hackrf rtl-sdr soapysdr rtaudio portaudio pandoc

git clone https://github.com/AlexandreRouma/SDRPlusPlus.git

cd [SDRPlusPlus](http://scutum.local:8888/#SDRPlusPlus) mkdir -p build cd build cmake .. \ -DOPT_BUILD_AUDIO_SINK=OFF \ -DOPT_BUILD_BLADERF_SOURCE=OFF \ -DOPT_BUILD_M17_DECODER=OFF \ -DOPT_BUILD_NEW_PORTAUDIO_SINK=ON \ -DOPT_BUILD_PLUTOSDR_SOURCE=OFF \ -DOPT_BUILD_PORTAUDIO_SINK=ON \ -DOPT_BUILD_SOAPY_SOURCE=ON \ -DOPT_BUILD_SDRPLAY_SOURCE=ON

make -j$(sysctl -n hw.ncpu)

sudo make install

sdrpp

---

That's what worked for me. Good luck.

Pascal

**Reply by AlexandreRouma** [reply]
those instructions are outdated, follow the CI build scripts instead

**Reply by mrkva** [reply]
Update: I downloaded the nightly Mac Intel build, it runs OK on the M1 and works with SDRPlay. There was no need to compile anything. I just need to start it from within the app (/SDR++.app/Contents/MacOS/sdrpp), otherwise it crashes.

**Comment by Terminator0007** [comment]
Hi. Today I've been trying to build SDR++ on my Mac mini M1 and installation stopped during built target network sink. Also ready packages from your repository for Mac with Intel CPU and for Mac M1 are stopped ( before my installation ). I can open app ( for Intel CPU ) but system do not want to start and recive from SDR source ( RTL-SDR, SDRPlay and from another SDR++ server ). I think after last update of OS Monterey some part of OS structure is changed by Apple .... And now I have SDR++ on shitty Raspberry Pi .....

[ 84%] Linking CXX shared library soapy_source.dylib
[ 84%] Built target soapy_source
1 warning generated.
[ 86%] Linking CXX shared library audio_sink.dylib
[ 86%] Built target audio_sink
1 warning generated.
[ 87%] Linking CXX shared library network_sink.dylib
[ 87%] Built target network_sink
make: *** [all] Error 2
